hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/StoreScanner.java

```diff
@@ -223,6 +224,10 @@ private void addCurrentScanners(List<? extends KeyValueScanner> scanners) {
     this.currentScanners.addAll(scanners);
   }
 
+  private static boolean isOnlyLatestVersionScan(Scan scan) {
+    return !scan.isRaw() && scan.getTimeRange().getMax() == HConstants.LATEST_TIMESTAMP;
+  }
```

Comment:
nit: perhaps good to comment that we don't need to check for Scan#getMaxVersions because live version file generated by dual file writer will anyways retain max versions specified in ColumnFamilyDescriptor for the given CF?